File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int,int>h;
        int n=nums.size();
        
        for(int i=0;i<n;i++){
            h[nums[i]]++;
        }
      for(auto it:nums){
      if(h[it]>1){
        return true;
      }
                        
       
        }
    
        return  false;
        
    }
    
};

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        int n=nums.length;
        HashSet<Integer> set=new HashSet<Integer>();
        for(int i:nums){
            set.add(i);
        }
        int m=set.size();
        if(n==m){
            return false;
        }
        return true;
    }
}